src/basic/process-util.c

```diff
@@ -1745,12 +1783,16 @@ int pidref_safe_fork_full(
         pid_t pid;
         int r, q;
 
-        assert(!FLAGS_SET(flags, FORK_WAIT));
-
         r = safe_fork_full(name, stdio_fds, except_fds, n_except_fds, flags, &pid);
-        if (r < 0)
+        if (r < 0 || !ret_pid)
                 return r;
 
+        if (r > 0 && FLAGS_SET(flags, FORK_WAIT)) {
+                /* If we are in the parent and successfully waited, then the process doesn't exist anymore */
+                *ret_pid = PIDREF_NULL;
+                return r;
+        }
```

Comment:
Hmm, what's the use case though?

Reply:
It's very useful in the test case later: we have nested forks, and it's good always getting a pidref to each, even if it's only useful inside the process...

will add comment.

but you are right, this is mostly syntactic sugar for my test, not more. But i think there's little reason to prohibit FORK_WAIT here if we can just make it work reasonably well.

After all in the long run we should probably try to get rid of all naked uses of pid_t and move to PidRef everywhere, including for all our own children. It's kinda weird if we couldn't use pidref_fork_safe() once with use FORK_WAIT...